### ISSUE TITLE
When testing it would be nice to over-ride the config file

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1229]
We do have some PRs where we discuss the future of the config file for run-time dependency injection, but unless we do remove all such options then it would be nice to over-ride the config file [mainly for me when switching between crates and astropy]

This could be done with a "command line" option for the test suite or using an environment variable. The latter would allow users to change the location, not just testers, but the former is "easier" to set up for a test.

**Comment by DougBurke** [comment]
With #1185 we can now does this for the io backend. @hamogu is working on making the plot backend replaceable (e.g. #1191). We may want to provide fixtures to make it easy to switch (currently there is kind-of one for plotting and I recently wrote some code for replacing the I/O backend which we might want to package up)

**Comment by hamogu** [comment]
Let's wait until #1185 is also in and then we can have one solution that works for both IO and plotting.

**Comment by DougBurke** [comment]
My main plan is to link up all the issues so we can solve them with a single PR ;-)

**Comment by DougBurke** [comment]
@hamogu I forget (and am too lazy to look), but does your new design support a user switching a backend at runtime?

A more practical use case for the general public than "for testing" is that I may have my default plot backend set to pylab but then I am in a jupyter notebook and want to switch to a bokeh/... backend for this sessino.

**Comment by DougBurke** [comment]
And the same process for plotting would also work for the I/O backend (in some not-yet-done PR)?

**Comment by hamogu** [comment]
It works for the IO backend due to the merged-several-months-ago #1185: https://sherpa.readthedocs.io/en/latest/overview/astro_io.html?#change-the-backend

**Comment by DougBurke** [comment]
I am closing this since we can set the `SHERPARC` environment file to select a different file - e.g. I use

```
% SHERPARC=sherpa/sherpa.rc pytest ...
% SHERPARC=sherpa/sherpa-standalone.rc pytest ...
```
